Standards/scs-0112-v1-sonic.md

```diff
@@ -0,0 +1,108 @@
+
+## Design Considerations
+
+There are different ways SONiC support can be implemented in SCS, very similar to existing approaches with Linux.
+
+### Options considered
+
+#### Option 1: SCS distribution of SONiC
+
+With this approach, SCS will create it's own distribution of SONiC, similar to what Debian or Arch are for Linux. This distribution will be based on the SONiC community distribution, but will have SCS specific modules, which will be developed and maintained by SCS. SCS will contribute its code to dedicated SCS repositories and build its own SONiC images. The code can eventually be pushed upstream, but not as top priority. This approach will allow SCS to have a clear path for SONiC support and integration in SCS, but will also require SCS to maintain a distribution of SONiC, which is a significant effort. Upstream/downstream changes will have to be managed and maintained. However the advantage is that SCS will have full control over the distribution and can make changes as needed. Users will have to use the SCS distribution of SONiC, which will be based on the community distribution. If users already deploy community or enterprise SONiC, a migration path to SCS SONiC will be needed.
+
+#### Option 2: SCS will support SONiC but will not change it
+
+SCS supports enterprise ans community versions of SONiC but will not develop its own code for it. This will significantly limit the ability to develop new features for SDN, because all changes will be done in the tooling around SONiC and not in the OS itself. The advantages are that SCS will still improve SONiC support and will have minimal effort for this. The downside is that some features like OVN control plane for SONiC will not be possible.
+
+#### Option 3: SCS develops SCS-specific modules as add-on for any SONiC (Community or Enterprise)
+
+In option 3, SCS will change SONiC by releasing its own modules for it. Those module can be provided as add-ons and installed on top of any version, community or enterprise. While compatability between the modules the SONiC releases will need to be maintained, there will be much broader support for SONiC and users will be able to pick and chose distributions based on their existing relationships and experience and use SCS independent of this. In cases where SCS provides contributions to core SONiC, those can be made in upstream Community repositories, so that the whole community including the propriatory vendors can adopt them eventually.
+
+#### Option 4: SCS does not adopt SONiC at all
+
+This option entails no dedicated effort on SCS's part in supporting SONiC network equipement for it's users and software stack. Users can still use SONiC from what is available by other projects or if they invest the effort themselves. This has several disadvantages:
+
+- SCS is not contributing to the SONiC community
```

Comment:
In which way is this a disadvantage to the SCS project or its participants?

Reply:
The idea was that SCS adopts OSS strategy and contributing and supporting other open-source project is seen as a good thing, also on non-technical level.